services/github/github-auth-service.js

```diff
@@ -2,21 +2,22 @@ import gql from 'graphql-tag'
 import { mergeQueries } from '../../core/base-service/graphql.js'
 import { BaseGraphqlService, BaseJsonService } from '../index.js'
 
-function createRequestFetcher(context, config) {
+function createRequestFetcher(context, config, neededScopes) {
   const { sendAndCacheRequestWithCallbacks, githubApiProvider } = context
 
   return async (url, options) =>
-    githubApiProvider.requestAsPromise(
-      sendAndCacheRequestWithCallbacks,
+    githubApiProvider.requestAsPromise({
+      request: sendAndCacheRequestWithCallbacks,
       url,
-      options
-    )
+      options,
+      neededScopes,
+    })
 }
 
 class GithubAuthV3Service extends BaseJsonService {
-  constructor(context, config) {
+  constructor(context, config, neededScopes) {
```

Comment:
This felt like the best way to allow service classes to enumerate their scope needs. Just like what's shown in the unit tests below, all that'd be needed for such classes (e.g. #4184) is an explicit constructor with a "hey i need this scope" flag. We could even extend the inheritance hierarchy with another set of class levels, e.g. `GitHubPackagesAuthV4 extends GithubAuthV4Service` with common scope needs so that individual service classes could extend and not have to worry about explicitly providing them